### PR TITLE
Save modified column for persist entity

### PR DESCRIPTION
**Pull request by stood**

### CODE REVIEW

File: sources/lib/Model/FlexibleEntity/ModifiedColumnEntityTrait.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * AddModifiedColumn
+     *
+     * @param $column
+     * @return FlexibleEntityInterface
+     */
+    public function addModifiedColumn($column)
+    {
+        if (!in_array($column, $this->modified_columns))
```
Comment (sanpii) [inline sources/lib/Model/FlexibleEntity/ModifiedColumnEntityTrait.php]: Missing braces (psr2).

File: sources/lib/Model/FlexibleEntity/ModifiedColumnEntityTrait.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * @return array
+     */
+    public function getModifiedColumns()
+    {
+        return $this->modified_columns;
+    }
+
+    /**
+     * AddModifiedColumn
```
Comment (sanpii) [inline sources/lib/Model/FlexibleEntity/ModifiedColumnEntityTrait.php]: Only return type is essential. Other comments is a duplicated (= bad) information.

File: sources/lib/Model/FlexibleEntity/ModifiedColumnEntityTrait.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * removeModifiedColumn
+     *
+     * @param $column
+     * @return FlexibleEntityInterface
+     */
+    public function removeModifiedColumn($column)
+    {
+        if (($key = array_search($column, $this->modified_columns, true)) !== FALSE)
```
Comment (sanpii) [inline sources/lib/Model/FlexibleEntity/ModifiedColumnEntityTrait.php]: Missing braces. Condition + affectation is not a goode idea for readability.

File: sources/lib/Model/FlexibleEntity.php [hunk 4]
```diff
@@ -99,6 +99,7 @@ public function __construct(array $values = null)
     {
         $this->container[$var] = $value;
         $this->touch();
+        $this->addModifiedColumn($var);
```
Comment (sanpii) [inline sources/lib/Model/FlexibleEntity.php]: Is `touch` and `addModifiedColumn` methods can be merged?

Comment (stood) [inline sources/lib/Model/FlexibleEntity.php]: It's possible but is it really good idea because it's a different functionnality ? ping @chanmix51

Comment (chanmix51) [inline sources/lib/Model/FlexibleEntity.php]: `touch` must be declared as abstract in the `ModifiedColumnEntityTrait`

**Pull request merged**